tests/components/mqtt/test_init.py

```diff
@@ -2806,43 +3036,50 @@ async def test_unload_config_entry(
     new_yaml_config_file.write_text(new_yaml_config)
     with patch.object(hass_config, "YAML_CONFIG_FILE", new_yaml_config_file):
         assert await hass.config_entries.async_unload(mqtt_config_entry.entry_id)
+        new_mqtt_config_entry = mqtt_config_entry
```

Comment:
MyPy does not like two assertions when it seems nothing is changed in the direct context